src/NUnitFramework/framework/Internal/Commands/SetUpTearDownItem.cs

```diff
@@ -92,11 +92,35 @@ private void RunSetUpOrTearDownMethod(TestExecutionContext context, IMethodInfo
             _methodValidator?.Validate(method.MethodInfo);
 
             var methodInfo = MethodInfoCache.Get(method);
+            RaiseStartedEvent(context, methodInfo);
 
-            if (methodInfo.IsAsyncOperation)
-                AsyncToSyncAdapter.Await(() => InvokeMethod(method, context));
-            else
-                InvokeMethod(method, context);
+            try
+            {
+                if (methodInfo.IsAsyncOperation)
+                    AsyncToSyncAdapter.Await(() => InvokeMethod(method, context));
+                else
+                    InvokeMethod(method, context);
+            }
+            finally
+            {
+                RaiseFinishedEvent(context, methodInfo);
+            }
+        }
+
+        private static void RaiseStartedEvent(TestExecutionContext context, MethodInfoCache.TestMethodMetadata methodInfo)
+        {
+            if (methodInfo.HasOneTimeSetUpAttribute)
+                context.ListenerExt.OneTimeSetUpStarted(context.CurrentTest);
+            else if (methodInfo.HasOneTimeTearDownAttribute)
+                context.ListenerExt.OneTimeTearDownStarted(context.CurrentTest);
+        }
+
+        private static void RaiseFinishedEvent(TestExecutionContext context, MethodInfoCache.TestMethodMetadata methodInfo)
+        {
+            if (methodInfo.HasOneTimeSetUpAttribute)
+                context.ListenerExt.OneTimeSetUpFinished(context.CurrentTest);
```

Comment:
The ListenerExt may not exist, thus be null, and if it is null, it should not be called. 
   context.ListenerExt?.OneTimeSetUpFinished(context.CurrentTest);

Reply:
Right. Updated in latest commit.